sanitycheck: Various fixes and enhancements:

nashif opened the pull request:
- Fixed --sub-test option
- add --list-test-duplicates to list duplicate identifiers

nashif committed:
sanitycheck: cleanup test inventory

Cleanup how we list testcases with --list-tests.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: add option to print duplicate identifiers

A new option to list all duplicate identifiers. This now will output the
following:

sanitycheck --list-test-duplicates

Tests with duplicate identifiers:
- arch.interrupt
  - tests/arch/arm/arm_interrupt/arch.interrupt
  - tests/kernel/interrupt/arch.interrupt
- system.settings.nffs.config_empty_lookups
  - tests/subsys/settings/nffs/base64/system.settings.nffs
  - tests/subsys/settings/nffs/raw/system.settings.nffs
- system.settings.nffs.config_insert
  - tests/subsys/settings/nffs/base64/system.settings.nffs
  - tests/subsys/settings/nffs/raw/system.settings.nffs
- system.settings.nffs.config_getset_unknown
  - tests/subsys/settings/nffs/base64/system.settings.nffs
  - tests/subsys/settings/nffs/raw/system.settings.nffs
- system.settings.nffs.config_getset_int
  - tests/subsys/settings/nffs/base64/system.settings.nffs
  - tests/subsys/settings/nffs/raw/system.settings.nffs
..
..
..

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: fix --sub-test option

This option was not working, make it work again using the new function
for managing test inventory.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

chen-png reviewed (approved):
it's fine.  fix the duplicate test case number reduce by sorted(set()) function.

nashif committed:
sanitycheck: report results before coverage generation

First report results, then start with coverage processing.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: enabling coverage should not imply --enable-slow

Those are two unrelated options.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: fix parsting tests from source files

Fixed bug in parsing testnames from source files. We added 1cpu testing
but did not update the regex and we have been passing wrong path to the
glob as well, meaning tests were not parsed at all.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
